SpamPkg/SpamBinBuild.py

```diff
@@ -249,15 +246,129 @@ def SetPlatformEnv(self):
-        ret = RunCmd(cmd, args)
-        return ret
+        build_output = Path(self.env.GetValue("BUILD_OUTPUT_BASE"))
+        mmsupv_dir = build_output / "X64" / "MmSupervisorPkg" / "Core" / "MmSupervisorCore" / "DEBUG"
+        stm_dir = build_output / "X64" / "SpamPkg" / "Core" / "Stm" / "DEBUG"
+
+        try:
+            # Calculate the hash of the MmSupervisorCore auxilary file
+            aux_file = mmsupv_dir / "MmSupervisorCore.aux"
+            aux_hash = calculate_aux_hash(aux_file)
+
+            # Find the offset of PcdAuxBinHash in the stm dll
+            stm_dir = build_output / "X64" / "SpamPkg" / "Core" / "Stm" / "DEBUG"
+            stm_map = stm_dir / "Stm.map"
+            stm_dll = stm_dir / "Stm.dll"
+            pcd = "PcdAuxBinHash"
+            pcd_addr = get_patch_pcd_address(stm_map, stm_dll, pcd)
+
+            if pcd_addr is None:
+                logging.error(f"Pcd {pcd} not found in {stm_map}.")
+                return -1
+
+            # Patch the hash value into the stm dll
+            patch_pcd_value(stm_dll, pcd_addr, aux_hash)
+        except FileNotFoundError as e:
+            logging.error(f"File {e} not found.")
+            return -1
+        except RuntimeError as e:
+            logging.error(e)
+            return -1
+
+        # Generate the stm binary
+        base_tools_dir = Path(os.environ['BASE_TOOLS_PATH'])
+        gen_stm = base_tools_dir / "Bin" / "Win32" / "GenStm.exe"
+        stm_dll = stm_dir / "Stm.dll"
+        stm_bin = stm_dir / "Stm.bin"
+        cmd = str(gen_stm)
+        args = f"-e --debug 5 {stm_dll} -o {stm_bin}"
+        return RunCmd(cmd, args)
+
+
+def calculate_aux_hash(file: Path) -> str:
+    """Calculates the hash of the given file.
+    
+    Returns:
+        str: The hex form of the hash as a string
+
+    Raises:    
+        FileNotFoundError: The file does not exist
+    """
+    if not file.exists():
+        raise FileNotFoundError(file)
+
+    hasher = hashlib.new("sha256")
+    with open(file, 'rb') as f:
+        while True:
+            data = f.read(65536)
+            if not data:
+                break
+            hasher.update(data)
+    return hasher.hexdigest()
+
+
+def get_patch_pcd_address(map: Path, dll: Path, pcd_name: str) -> int:
+    """Gets the address of the given PCD in the patch PCD table.
+
+    Args:
+        file (Path): the path to the patch PCD table
+        pcd_name (str): the name of the PCD to find
+
+    Returns:
+        int: the address of the PCD in the patch PCD table
+        None: The pcd does not exist in the table
+    
+    Raises:
+        FileNotFoundError: if map or dll do not exist
+        RuntimeError: if GenPatchPcdTable fails
+    """
+    if not map.exists():
+        raise FileNotFoundError(map)
+    if not dll.exists():
+        raise FileNotFoundError(dll)
+
+    cmd = "GenPatchPcdTable"
+    args = f"-m {map} -e {dll}"
+    RunCmd(cmd, args)
+
+    pcd_table = map.with_suffix(".BinaryPcdTable.txt")

```

Comment:
Should we have a txt file that has a list of supported aux files somewhere?

Reply:
What do you mean supported aux files? The pcd_table variable you put this comment on is just an artifact that the edk2 maintained tool `GenPatchPcdTable`, so that name is hardcoded.

If you mean the actual aux file, It should be such that any aux file generated by `aux_gen` should be supported on the firmware side.